### PR TITLE
cpu: add and use shared code for timer_set()

### PR DESCRIPTION
**Pull request by haukepetersen**
how about a little less code duplication?! here we go.

### CODE REVIEW

File: drivers/periph_common/timer.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @}
+ */
+
+#include "periph/timer.h"
+
+#ifndef PERIPH_TIMER_PROVIDES_SET
+int timer_set(tim_t dev, int channel, unsigned int timeout)
+{
+    return timer_set_absolute(dev, channel, timer_read(dev) + timeout);
```
Comment (aabadie) [inline drivers/periph_common/timer.c]: I might also be good to add a safety check on dev before this line. Something like:
```c
if (dev >= TIMER_NUMOF) {
        return -1;
    }
```
I saw that in one of the other impl that is removed in this PR.

Comment (haukepetersen) [inline drivers/periph_common/timer.c]: not needed, this is done in timer_set_absolute.

And it should not be done using the snipped you posted above, but by using `assert()`...

**Comment by smlng** [comment]
looks good in general, I'll try to take all boards I've available for a test the next days - just to make sure.

**Comment by haukepetersen** [comment]
@aabadie, @smlng any updates on this?

**Comment by gebart** [comment]
ping @aabadie

**Comment by kYc0o** [comment]
@aabadie is in vacations. I think it's safe to dismiss his review.

**Comment by kYc0o** [comment]
I'd like to run Murdock again, it looked outdated.

**Pull request merged**